src/infra/datasets/postgres/src/repos/postgres_dataset_entry_repository.rs

```diff
@@ -95,7 +98,7 @@ impl DatasetEntryRepository for PostgresDatasetEntryRepository {
                     dataset_name as name,
                     created_at   as "created_at: _"
                 FROM dataset_entries
-                ORDER BY dataset_name ASC
+                ORDER BY created_at ASC
```

Comment:
Hmm, why like this?
Probably the order will need to be parameterizable later.

Reply:
was based on this logic: 

- Let's imagine that a new `dataset_entry` was added during streaming
- When sorting by `created_at`, the new entry will be guaranteed to be on the last page, without affecting the previous ones in any way
- When sorting by name, a new entry may be inserted in the middle of an already issued page, creating a shift and duplicates

Reply:
I understand the phantom reads are possible, but have you noticed this actually?

Reply:
No, I've seen it in practice -- more of an attempt to avoid potential problems